internal/otel/oteltest/collector.go

```
@@ -0,0 +1,69 @@
+	collpb.UnimplementedMetricsServiceServer
+
+	mu   sync.Mutex
+	data []*mpb.ResourceMetrics
+
+	listener net.Listener
+	server   *grpc.Server
+}
+
+// NewGRPCCollector listens on an ephemeral localhost port.
+func NewGRPCCollector() (*GRPCCollector, error) {
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)